### DISCUSSION TITLE
How does he do it?

### DISCUSSION THREAD

**Discussion by g-l-i-t-c-h-o-r-s-e** [typekcz/beepboop-steam#14] [General]
I'm curious and would be ever so grateful if you could explain how you're even able get it to join voice chat?
Normally, trying to join steam group voice chats in any web browser, results in failure and error codes such as:
`(WebRTC) setRemoteDescription error: OperationError: Description contains duplicate extension id 1 on level 0`
or
`(WebRTC) Got failure (2) response from InitiateWebRTCConnection`

**Comment by typekcz** [comment]
Well, currently it doesn't work.
It worked in the past, but then Chrome stopped supporting WebRTC Plan B, which is what Valve is using for the web Steam chat.
I think I tried running it on older Chromium, but without success (probably remote kill switch like with Flash?)
Then I said I'll wait until Valves changes implementation to use WebRTC Unified Plan, but months went by and still nothing.
Steam voice chat on the web is just broken 😢

**Reply by g-l-i-t-c-h-o-r-s-e** [reply]
just automate the steam client instead lol, maybe with selenium or just DevTools
`steam.exe -cef-enable-debugging -console -dev`
<details> 
  <summary>spoiler alert</summary>
   steam is a chromium browser lol, on port 8080
</details>

**Comment by typekcz** [comment]
Running full Steam on servers without GUI might be problematic...
But maybe using just CEF somehow could work?

**Reply by g-l-i-t-c-h-o-r-s-e** [reply]
Once you connect to CEF/debug port you don't really need a GUI unless you need to see what you're doing, but ye you can easily run js in CEF mode

**Comment by typekcz** [comment]
Yeah, but Steam itself probably won't run without xserver. Also you need to login somehow.
Currently my deploy target is Heroku or no gui servers in general.

**Comment by g-l-i-t-c-h-o-r-s-e** [comment]
I wonder if this would work for the WebRTC Plan B issue
https://github.com/jitsi/sdp-interop

**Comment by typekcz** [comment]
Can't get it to work. With this it's not in the connecting loop, but immediatelly disconnects and I'm not getting any errors.
I look at the CEF, looks like Steam is using version 84 (2 years old!!!), so I tried downloading that version of Chromium and running bot on that. Same result...
Currently I'm not getting any errors at all, even without this hack, so I don't know what's going on anymore...
Anyway thanks for trying to help out.

Ok, one more thing came to my mind, download page source code from Steam client version - it's probably different that what is beiing served to browsers. I'll try tomorrow or some day.

**Comment by typekcz** [comment]
Hey, if you are still watching this thread, I wanted to let you know that I'm working new version with dockerized Steam client.
I have working version in develop branch, I'm still working on it. Not all features are working yet, but it should be done soon.
